tool/src/org/antlr/v4/codegen/target/CppTarget.java

```diff
@@ -117,4 +117,9 @@ public String getBaseVisitorFileName(boolean header) {
 		String listenerName = gen.g.name + "BaseVisitor";
 		return listenerName+extST.render();
 	}
+
+	@Override
+	public String translateThisAccessor(String code) {
+		return code.replaceAll("this\\.", "this->");
```

Comment:
Sorry, but I can't accept the soltuion based on string replace technique. It doesn't consider string literals and inefficient.

Reply:
Do you have any idea to solve the issue as below description?

Reply:
I have but I afraid it doesn't fit into ANTLR development roadmap.